src/hooks/useInterval.ts

```diff
@@ -1,27 +1,50 @@
-import { useEffect, useRef } from 'react'
+import { useEffect, useRef, useState, useCallback } from 'react'
+import isUndefinedOrNull from 'utils/isUndefinedOrNull'
+import useLastUpdated from './useLastUpdated'
 
-export default function useInterval(callback: () => void, delay: null | number, leading = true) {
-  const savedCallback = useRef<() => void>()
+export default function useInterval(callback: () => void, delay: undefined | null | number, leading = true) {
+  const [runImmediate, setRunImmediate] = useState(leading)
+  const [runAfter, setRunAfter] = useState(delay)
+  const savedCallback = useRef<() => void>(callback)
+  const [isReadyForUpdate, setIsReadyForUpdate] = useState(false)
+  const { lastUpdated, setLastUpdated: refresh } = useLastUpdated()
+
+  // Use props indirectly to not render set up timeout effect twice if both props changed.
+  useEffect(() => {
+    setRunImmediate(leading)
+    setRunAfter(delay)
+  }, [leading, delay])
```

Comment:
<img width="899" alt="Screen Shot 2022-07-01 at 08 45 04" src="https://user-images.githubusercontent.com/97418926/176806147-89cf5ebd-a264-4433-b3ee-7309cefb56da.png">


Could you check if this pattern is similar to this one? 

Reference: https://beta-reactjs-org-git-you-might-not-fbopensource.vercel.app/learn/you-might-not-need-an-effect

Reply:
In our case delay and leading props not used directly, so we can use it in effect without side effects